src/formula_ast.cc

```diff
@@ -248,8 +258,13 @@ double FormulaAst::evaluate(const std::vector<Variable::Type>& values, const std
         case BinaryOp::Div: return left / right;
         case BinaryOp::Times: return left * right;
         case BinaryOp::Pow: return std::pow(left, right);
+        case BinaryOp::Atan2: return std::atan2(left, right);
+        case BinaryOp::Max: return std::max(left, right);
+        case BinaryOp::Min: return std::min(left, right);
       };
     }
+    case NodeType::Undefined:
```

Comment:
Would it make more sense to remove the default and the `Undefined` type and let the compiler ensure the switch is exhaustive? I guess just the root node is that type.

Reply:
> I guess just the root node is that type

sorry I don't get this part, can you elaborate?

Reply:
Ah sorry, it isn't. I just looked at where `NodeType::Undefined` is used and it appears to just be the default constructor, which I think is never actually used.

Reply:
ok, so I'll just remove the Undefined case :+1:

Reply:
done